python/ray/ml/checkpoint.py

```
@@ -368,8 +416,27 @@ def as_directory(self) -> Iterator[str]:
             yield self._local_path
         else:
             temp_dir = self.to_directory()
+            del_lock_path = _get_del_lock_path(temp_dir)
             yield temp_dir
-            shutil.rmtree(temp_dir, ignore_errors=True)
+
+            # Cleanup
+            try:
+                os.remove(del_lock_path)
+            except Exception:
+                pass
```

logger.exception()? Best never to swallow exceptions.